include/storage/shared_datatype.hpp

```diff
@@ -115,9 +92,59 @@ class DataLayout
         }
     }
 
+    virtual inline void *GetBlockPtr(char *shared_memory, const std::string &name) const = 0;
+
+    std::map<std::string, Block> blocks;
```

Comment:
Can this be an unordered map?

Reply:
Funnily enough no - there's calling code that depends on `std::for_each(blocks.begin(), blocks.end())` iterating over the keys in the same order every time.

I feel like the concept of this class has been somewhat damaged by introducing mmapping, and a refactoring is in order.